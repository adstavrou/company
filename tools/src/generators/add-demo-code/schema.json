{
  "$schema": "http://json-schema.org/schema",
  "title": "Add Demo Generator",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The folder name for the new demo (e.g. buttons-demo)"
    },
    "title": {
      "type": "string",
      "description": "The display title for the demo (e.g. Buttons Demo)"
    }
  }
}