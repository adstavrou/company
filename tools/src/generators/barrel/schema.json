{
    "$schema": "http://json-schema.org/schema",
    "title": "Barrel Generator",
    "type": "object",
    "properties": {
        "project": {
            "type": "string",
            "description": "The name of the project (library) to generate the barrel for"
        }
    },
    "required": [
        "project"
    ]
}