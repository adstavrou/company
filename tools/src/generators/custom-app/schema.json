{
  "$schema": "http://json-schema.org/schema",
  "title": "Custom App Generator",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the app"
    }
  }
}