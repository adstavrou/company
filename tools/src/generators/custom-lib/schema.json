{
    "$schema": "http://json-schema.org/schema",
    "title": "Custom Lib Generator",
    "type": "object",
    "properties": {
        "name": {
            "type": "string",
            "description": "Library name"
        },
        "build": {
            "type": "boolean",
            "description": "Add build target?"
        },
        "unitTest": {
            "type": "boolean",
            "description": "Add unit tests?"
        }
    },
    "required": []
}