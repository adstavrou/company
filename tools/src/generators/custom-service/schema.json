{
    "$schema": "http://json-schema.org/schema",
    "title": "Custom Service Generator",
    "type": "object",
    "properties": {
      "name": {
        "type": "string",
        "description": "Name of the service"
      },
      "directory": {
        "type": "string",
        "description": "Target directory inside libs/"
      }
    },
    "required": []
  }