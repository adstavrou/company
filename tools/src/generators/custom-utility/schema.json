{
    "$schema": "http://json-schema.org/schema",
    "title": "Custom Utility Generator",
    "type": "object",
    "properties": {
        "name": {
            "type": "string",
            "description": "The name of the utility function"
        },
        "directory": {
            "type": "string",
            "description": "The library folder (e.g. shared, core-utils)"
        },
        "withTest": {
            "type": "boolean",
            "description": "Generate test file"
        }
    },
    "required": []
}