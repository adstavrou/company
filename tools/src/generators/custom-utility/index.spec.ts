import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { Tree } from '@nx/devkit';
import generator from './index';
import { CustomUtilitySchema } from './schema';

describe('custom-utility generator', () => {
  let tree: Tree;

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace();
    tree.write('libs/shared/src/index.ts', ''); // simulate existing lib structure
  });

  it('should generate a util file in the correct location', async () => {
    const options: CustomUtilitySchema = {
      name: 'slugify-title',
      directory: 'shared',
      withTest: false,
    };

    await generator(tree, options);

    const utilPath = 'libs/shared/src/slugify-title.util.ts';
    expect(tree.exists(utilPath)).toBe(true);

    const content = tree.read(utilPath)?.toString('utf-8');
    expect(content).toContain('export function slugifyTitle()');
    expect(content).toContain('// TODO: Implement slugifyTitle');
  });

  it('should generate a spec file if withTest is true', async () => {
    const options: CustomUtilitySchema = {
      name: 'format-price',
      directory: 'shared',
      withTest: true,
    };

    await generator(tree, options);

    const specPath = 'libs/shared/src/format-price.util.spec.ts';
    expect(tree.exists(specPath)).toBe(true);

    const specContent = tree.read(specPath)?.toString('utf-8');
    expect(specContent).toContain("describe('formatPrice'");
    expect(specContent).toContain('expect(formatPrice()');
  });

  it('should generate utility file and match snapshot', async () => {
    const options: CustomUtilitySchema = {
      name: 'hello world',
      directory: 'shared',
      withTest: true,
    };

    await generator(tree, options);

    const filePath = 'libs/shared/src/hello-world.util.ts';
    const fileContent = tree.read(filePath, 'utf-8');
    expect(fileContent).toMatchSnapshot();

    const testFilePath = 'libs/shared/src/hello-world.util.spec.ts';
    const testContent = tree.read(testFilePath, 'utf-8');
    expect(testContent).toMatchSnapshot();
  });
});
