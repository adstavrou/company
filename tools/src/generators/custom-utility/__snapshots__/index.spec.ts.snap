// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom-utility generator should generate utility file and match snapshot 1`] = `
"export function helloWorld() {
  // TODO: Implement helloWorld
}
"
`;

exports[`custom-utility generator should generate utility file and match snapshot 2`] = `
"import { helloWorld } from './hello-world.util';

describe('helloWorld', () => {
  it('should work', () => {
    expect(helloWorld()).toBeDefined();
  });
});
"
`;
